doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+
+* Starts with `/**` and ends with the matching `*/`.
+* Block comments can be nested (e.g., `\*\* \* inner */ */`). Documentation tools must respect this, ensuring that the comment block only ends at the closing  \*/ that matches the opening delimiter.
+* The content within the delimiters is treated as the documentation.
+* Leading whitespace followed by a single asterisk and a single following space on subsequent lines are considered stylistic and are stripped by doc generators.
+
+**Example**
+
+```
+/**
+ *
+ *
+ */
+```
+
+### **2.3. Content Format (Markdown)**
+
+The text within a documentation comment block is parsed as Markdown, allowing for rich text formatting. This includes headings, lists, code blocks, and emphasis, which are converted for instance to HTML in the generated documentation.
+
+### **2.4. References**
+
+A reference is a special directive within the Markdown content that creates a hyperlink to a Dart element. It is written by enclosing a Dart identifier in square brackets (e.g., `[foo]`).  See [Section 4](#4.-referenceable-elements) for detailed information about which elements can be referenced.
+
+Conceptually, these behave like [reference-style links](https://www.markdownguide.org/basic-syntax/#reference-style-links) in Markdown. The documentation generator resolves the name against the available source code to create the link's destination. See [Section 5](#5.-reference-lookup-and-resolution) for detailed resolution rules.
+
+## **3\. Placement of Documentation Comments**
+
+Doc comments are associated with the declaration that immediately follows them. They are only considered valid when placed directly before the following types of declarations:
+
+### **3.1. Directives**
+
+* `library` directives.
+
+### **3.2. Top-Level Declarations**
+
+* `class`
+* `mixin`
+* `enum`
+* `extension`
+* `extension type`
+* `typedef`
+* Top-level functions
+* Top-level variables
+* Top-level getters or setters (See [Section 6.2.2](#6.2.2.-getters-and-setters) for details)
+
+### **3.3. Member Declarations**
+
+* Constructors (including factory and named constructors)
+* Methods (instance, static)
+* Operators
+* Fields (instance, static)
+* Getters or setters (Not both, prefer getter)
+
+### **3.4. Enum Constants**
+
+* A doc comment can be placed before an individual enum value.
+
+### **3.5. Invalid Placements**
+
+While not strictly disallowed by the language, any other placement of a comment with the /// syntax, is not considered a doc comment, and hence should be ignored by documentation tools.
+
+## **4\. Referenceable Elements**
+
+A reference in a doc comment (e.g., `[name]`) can link to any Dart element that is visible from the Dart scope of the documented element. See [Section 5](#5.-reference-lookup-and-resolution) for more details about scoping.   This includes:
+
+### **4.1. Types**
+
+* Classes (e.g., `[MyClass]`)
+* Mixins (e.g., `[MyMixin]`)
+* Enums (e.g., `[MyEnum]`)
+* Named extensions (e.g., `[MyExtension]`)
```

Comment:
Nit: Not a type. Another reason to just use "top-level declarations". 😉

Reply:
done